controllers/profiles/status_enricher.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profiles
+
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+)
+
+// newStatusEnricher see defaultObjectEnsurer
+func newStatusEnricher(client client.Client, config *rest.Config, logger *logr.Logger, enricher statusEnricherFn) *statusEnricher {
```

Comment:
new, is this a pattern that was used?

Reply:
Yes, for object creators in Go it is a de-facto convention using the New or new keyword that you can see in https://go.dev/doc/effective_go